py/sentry_streams/runner.py

```diff
@@ -32,56 +73,23 @@ def main() -> None:
         "topics": {
             "logical-events": "events",
             "transformed-events": "transformed-events",
-        }
+        },
+        "broker": "localhost:9092",
     }
 
-    # def recurse_edge(input_name: str, stream: Any) -> None:
-    #     for next_step_name in p.edges.get(input_name, ()):
-    #         print(f"Apply step: {next_step_name}")
-    #         next_step: WithInput = cast(WithInput, p.steps[next_step_name])
-    #         recurse_edge(
-    #             next_step_name, next_step.apply_edge(stream, environment_config)
-    #         )
-
-    # for source in p.sources:
-    #     print(f"Apply source: {source.name}")
-    #     env_source = source.apply_source(env, environment_config)
-    #     recurse_edge(source.name, env_source)
-
-    def iterate_edges(step_streams: dict[str, Any]) -> None:
-        while step_streams:
-            for input_name in list(step_streams):
-                output_steps = p.outgoing_edges[input_name]
-                input_stream = step_streams.pop(input_name)
-
-                if not output_steps:
-                    continue
-
-                # check if the inputs are fanning out
-                if len(output_steps) > 1:
-                    pass
-
-                else:
-                    output_step_name = output_steps.pop()
-
-                    # check if the inputs are fanning in
-                    if len(p.incoming_edges[output_step_name]) > 1:
-                        pass
+    # This will not be harcdoded in the future
+    runtime_config: StreamAdapter = FlinkAdapter(environment_config, env)
+    translator = RuntimeTranslator(runtime_config)
 
-                    # 1:1 between input and output stream
-                    else:
-                        next_step: WithInput = cast(WithInput, p.steps[output_step_name])
-                        print(f"Apply step: {next_step.name}")
-                        output_stream = next_step.apply_edge(input_stream, environment_config)
-                        step_streams[next_step.name] = output_stream
+    pipeline.set_translator(translator)
 
     step_streams = {}
 
-    for source in p.sources:
+    for source in p_graph.sources:
         print(f"Apply source: {source.name}")
-        env_source = source.apply_source(env, environment_config)
+        env_source = source.apply_source()
         step_streams[source.name] = env_source
-        iterate_edges(step_streams)
+        iterate_edges(step_streams, p_graph)
 
     # submit for execution
     env.execute()
```

Comment:
Shouldn't the transaltor return an `env` object as that is Flink specific ?

It owuld be alright if we kept the runtime translation for a separate PR

Reply:
It seems that with Flink, all you need to do is register the source with the `env`. (We do that in the `FlinkAdapter`) 

Then, subsequently traversing the graph and translating each step just yields a new DataStream after each step. You don't need to pass around`env` as you traverse the graph. 

All you need is a call to `env.execute()` to actually submit the stream to be run.